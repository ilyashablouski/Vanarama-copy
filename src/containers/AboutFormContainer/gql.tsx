import { useMutation, useQuery, gql } from '@apollo/client';
import {
  CreateUpdatePersonMutation,
  CreateUpdatePersonMutationVariables,
} from '../../../generated/CreateUpdatePersonMutation';
import {
  GetAboutYouDataQuery,
  GetAboutYouDataQueryVariables,
} from '../../../generated/GetAboutYouDataQuery';
import { GetAboutYouPageQuery } from '../../../generated/GetAboutYouPageQuery';
import AboutForm from '../../components/AboutForm';

export const CREATE_UPDATE_PERSON = gql`
  mutation CreateUpdatePersonMutation($input: PersonInputObject!) {
    createUpdatePerson(input: $input) {
      ...AboutFormPerson
    }
  }
  ${AboutForm.fragments.person}
`;

export const GET_ABOUT_YOU_DATA = gql`
  query GetAboutYouDataQuery($uuid: ID!) {
    personByUuid(uuid: $uuid) {
      ...AboutFormPerson
    }
  }
  ${AboutForm.fragments.person}
`;

export const GET_ABOUT_YOU_PAGE_DATA = gql`
  query GetAboutYouPageQuery {
    allDropDowns {
      ...AboutFormDropdownData
    }
  }
  ${AboutForm.fragments.dropdownData}
`;

export function useAboutPageDataQuery() {
  return useQuery<GetAboutYouPageQuery>(GET_ABOUT_YOU_PAGE_DATA, {
    fetchPolicy: 'no-cache',
  });
}

export function useAboutYouData(personByUuid?: string) {
  return useQuery<GetAboutYouDataQuery, GetAboutYouDataQueryVariables>(
    GET_ABOUT_YOU_DATA,
    {
      variables: {
        /**
         * If there's not a personUuid then we can set it to anything
         * as it will be excluded from the query anyway
         */
        uuid: personByUuid || '🐔',
      },
      fetchPolicy: 'no-cache',
      skip: !personByUuid,
    },
  );
}

export function useCreatePerson(
  onCompleted?: (data: CreateUpdatePersonMutation) => void,
) {
  return useMutation<
    CreateUpdatePersonMutation,
    CreateUpdatePersonMutationVariables
  >(CREATE_UPDATE_PERSON, {
    onCompleted,
  });
}
