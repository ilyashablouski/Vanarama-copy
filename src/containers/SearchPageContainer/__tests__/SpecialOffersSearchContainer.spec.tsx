import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import React from 'react';
import preloadAll from 'jest-next-dynamic';
import { act, render, screen, waitFor } from '@testing-library/react';
import { VehicleTypeEnum } from '../../../../generated/globalTypes';
import { useVehiclesList } from '../gql';
import { GET_SEARCH_POD_DATA } from '../../SearchPodContainer/gql';
import { GET_PRODUCT_CARDS_DATA } from '../../CustomerAlsoViewedContainer/gql';
import SpecialOffersSearchContainer from '../SpecialOffersSearchContainer';

const metaData = {
  title: 'Car Leasing Deals | Personal & Business Contract Hire | Vanarama',
  name: 'HubCarPage',
  metaRobots: 'all',
  metaDescription:
    'Find unbeatable Car Leasing Deals at Vanarama. Get top personal & business lease offers on brand new, in-stock cars in every make and model. Save money and lease your dream car today.',
  legacyUrl: 'https://www.vanarama.com/car-leasing.html',
  pageType: null,
  canonicalUrl: 'https://www.vanarama.com/car-leasing.html',
  slug: 'car-leasing',
  schema: {
    '@graph': [
      {
        '@type': 'FAQPage',
        mainEntity: [
          {
            name: 'Why Lease A Car?',
            '@type': 'Question',
            acceptedAnswer: {
              text:
                'Leasing your brand new car with Vanarama means you can drive the car of your dreams at a price you can afford. And you can upgrade to a new one every few years',
              '@type': 'Answer',
            },
          },
          {
            name: 'How Does Car Leasing Work?',
            '@type': 'Question',
            acceptedAnswer: {
              text:
                "Personal car leasing works very much like renting, but for a longer time period. You choose your new car, the contract length, how much you'd like your initial rental to be and how many miles you'll drive over the period. At the end of your lease agreement, you return the car & upgrade",
              '@type': 'Answer',
            },
          },
        ],
      },
      {
        url: 'https://www.vanarama.com/car-leasing.html',
        name: 'Vanarama',
        '@type': 'Service',
        '@context': 'http://schema.org',
        description:
          'Leasing your brand new car with Vanarama means you can drive the car of your dreams at a price you can afford.',
        serviceType: 'Car Leasing',
      },
    ],
  },
  publishedOn: null,
  breadcrumbs: [
    {
      href: '/',
      label: 'Home',
    },
    {
      label: 'Car Leasing',
    },
  ],
  __typename: 'Meta',
};
const mockData = {
  loading: false,
  refetch() {
    return this.data;
  },
  data: {
    productCard: [
      {
        vehicleType: VehicleTypeEnum.CAR,
        capId: '83615',
        manufacturerName: 'manufacturerName',
        rangeName: 'rangeName',
        derivativeName: 'derivativeName',
        averageRating: 4.5,
        isOnOffer: true,
        offerPosition: 5,
        leadTime: '',
        imageUrl: '',
        keyInformation: [],
        businessRate: 55,
        personalRate: 55,
      },
    ],
    derivatives: [
      {
        id: '83615',
        derivativeName: '1.0 EcoBoost 125 ST-Line Nav 5dr',
        slug: '10-ecoBoost-125-st-line-nav-5dr',
        capCode: 'capCode',
        name: 'name',
        manufacturer: {
          name: 'Ford',
          slug: 'ford',
        },
        model: {
          name: 'Focus',
          slug: 'focus',
        },
        fuelType: {
          name: 'name',
        },
        transmission: {
          name: 'name',
        },
        bodyStyle: {
          name: 'Hatchback',
        },
        range: {
          name: 'Focus',
          slug: 'focus',
        },
        __typename: 'derivative',
      },
    ],
  },
  error: undefined,
};

const filterListResponse = {
  vehicleTypes: [VehicleTypeEnum.CAR],
  groupedRangesWithSlug: [
    {
      parent: { label: 'Citroën', slug: 'Citroën' },
      children: [
        { label: 'Berlingo', slug: 'Berlingo' },
        { label: 'Dispatch', slug: 'Dispatch' },
        { label: 'Relay', slug: 'Relay' },
      ],
    },
    {
      parent: { label: 'Dacia', slug: 'Dacia' },
      children: [{ label: 'Duster', slug: 'Duster' }],
    },
    {
      parent: { label: 'BMW', slug: 'BMW' },
      children: [
        { label: '3 series', slug: '3 series' },
        { label: '4 series', slug: '4 series' },
      ],
    },
  ],
  bodyStyles: ['Dropside Tipper', 'Large Van'],
  transmissions: ['Automatic', 'Manual'],
  fuelTypes: ['diesel', 'iii'],
};

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
    rewrite: jest.fn(),
    pathname: '/car-leasing',
    query: {},
    route: '/car-leasing',
    asPath: '/car-leasing',
  }),
}));

jest.mock('../gql', () => ({
  ...jest.requireActual('../gql'),
  useVehiclesList: jest.fn(),
}));

// ARRANGE
let filterMockCalled = false;

(useVehiclesList as jest.Mock).mockReturnValue([
  jest.fn(),
  {
    data: {
      vehicleList: {
        totalCount: 1,
        pageInfo: {
          startCursor: 'MQ',
          endCursor: 'OQ',
          hasNextPage: false,
          hasPreviousPage: false,
        },
        edges: [
          {
            cursor: 'MQ',
            node: {
              vehicleType: VehicleTypeEnum.CAR,
              offerRanking: 1,
              onOffer: true,
              derivativeId: '83615',
              capCode: 'FOFO10TN55HPTM  6   ',
              manufacturerName: 'Ford',
              modelName: 'Focus Hatchback',
              derivativeName: '1.0 EcoBoost 125 ST-Line Nav 5 Doors',
              bodyStyle: 'Hatchback',
              transmission: 'Manual',
              fuelType: 'Petrol',
              financeProfiles: [
                {
                  leaseType: 'PERSONAL',
                  rate: 210.96,
                  term: 24,
                  upfront: 9,
                  upfrontPayment: 1898.64,
                  mileage: 6000,
                  maintained: false,
                },
                {
                  leaseType: 'BUSINESS',
                  rate: 175.96,
                  term: 24,
                  upfront: 9,
                  upfrontPayment: 1583.64,
                  mileage: 6000,
                  maintained: false,
                },
              ],
            },
          },
        ],
      },
    },
  },
]);

const mocksResponse: MockedResponse[] = [
  {
    request: {
      query: GET_SEARCH_POD_DATA,
      variables: {
        vehicleTypes: [VehicleTypeEnum.CAR],
        onOffer: true,
        bodyStyles: [],
        fuelTypes: [],
      },
    },
    result: () => {
      filterMockCalled = true;
      return {
        data: {
          filterList: filterListResponse,
        },
        refetch: jest.fn(),
      };
    },
  },
  {
    request: {
      query: GET_SEARCH_POD_DATA,
      variables: {
        vehicleTypes: [VehicleTypeEnum.CAR],
        onOffer: true,
        fuelTypes: [],
        bodyStyles: [],
        transmissions: ['Automatic'],
      },
    },
    result: () => {
      filterMockCalled = true;
      return {
        data: {
          filterList: filterListResponse,
        },
        refetch: jest.fn(),
      };
    },
    newData: jest.fn(() => ({
      data: {
        filterList: {
          vehicleTypes: [VehicleTypeEnum.CAR],
          groupedRanges: [
            {
              parent: 'Citroën',
              children: ['Berlingo', 'Dispatch'],
            },
            {
              parent: 'Dacia',
              children: ['Duster'],
            },
          ],
          bodyStyles: ['Dropside Tipper', 'Large Van'],
          transmissions: ['Automatic', 'Manual'],
          fuelTypes: ['diesel', 'iii', 'electro'],
        },
      },
      refetch: jest.fn(),
    })),
  },
  {
    request: {
      query: GET_SEARCH_POD_DATA,
      variables: {
        vehicleTypes: [VehicleTypeEnum.CAR],
        onOffer: true,
        fuelTypes: [],
        bodyStyles: [],
        transmissions: ['Automatic'],
      },
    },
    result: () => {
      filterMockCalled = true;
      return {
        data: {
          filterList: filterListResponse,
        },
      };
    },
    newData: () => {
      filterMockCalled = true;
      return {
        data: {
          filterList: {
            vehicleTypes: [VehicleTypeEnum.CAR],
            groupedRanges: [
              {
                parent: 'Citroën',
                children: ['Berlingo', 'Dispatch'],
              },
              {
                parent: 'Dacia',
                children: ['Duster'],
              },
            ],
            bodyStyles: ['Dropside Tipper', 'Large Van'],
            transmissions: ['Automatic', 'Manual'],
            fuelTypes: ['diesel', 'iii', 'electro'],
          },
        },
      };
    },
  },
  {
    request: {
      query: GET_PRODUCT_CARDS_DATA,
      variables: {
        capIds: ['83615'],
        vehicleType: VehicleTypeEnum.CAR,
      },
    },
    result: () => {
      return {
        data: {
          productCard: {
            vehicleType: VehicleTypeEnum.CAR,
            capId: '83615',
            manufacturerName: 'manufacturerName',
            rangeName: 'rangeName',
            derivativeName: 'derivativeName',
            averageRating: 4.5,
            isOnOffer: false,
            offerPosition: 5,
            leadTime: '',
            imageUrl: '',
            keyInformation: [{ name: 'Transmission', value: 'Manual' }],
            businessRate: 55,
            personalRate: 55,
          },
          derivatives: mockData.data.derivatives,
          vehicleList: {},
        },
        refetch: jest.fn(),
      };
    },
  },
  {
    request: {
      query: GET_PRODUCT_CARDS_DATA,
      variables: {
        capIds: [],
        vehicleType: VehicleTypeEnum.CAR,
      },
    },
    result: () => {
      return {
        data: {
          productCard: {},
          derivatives: [],
          vehicleList: {},
        },
        refetch: jest.fn(),
      };
    },
  },
  {
    request: {
      query: GET_PRODUCT_CARDS_DATA,
      variables: {
        capIds: ['83615'],
        vehicleType: VehicleTypeEnum.CAR,
      },
    },
    result: () => {
      return {
        data: {
          productCard: [
            {
              vehicleType: VehicleTypeEnum.CAR,
              capId: '836151',
              manufacturerName: 'manufacturerName',
              rangeName: 'rangeName',
              derivativeName: 'derivativeName',
              averageRating: 4.5,
              isOnOffer: false,
              offerPosition: 5,
              leadTime: '',
              imageUrl: '',
              keyInformation: [],
              businessRate: 55,
              personalRate: 55,
            },
            {
              vehicleType: VehicleTypeEnum.CAR,
              capId: '836152',
              manufacturerName: 'manufacturerName',
              rangeName: 'rangeName',
              derivativeName: 'derivativeName',
              averageRating: 4.5,
              isOnOffer: false,
              offerPosition: 5,
              leadTime: '',
              imageUrl: '',
              keyInformation: [],
              businessRate: 55,
              personalRate: 55,
            },
            {
              vehicleType: VehicleTypeEnum.CAR,
              capId: '836153',
              manufacturerName: 'manufacturerName',
              rangeName: 'rangeName',
              derivativeName: 'derivativeName',
              averageRating: 4.5,
              isOnOffer: false,
              offerPosition: 5,
              leadTime: '',
              imageUrl: '',
              keyInformation: [],
              businessRate: 55,
              personalRate: 55,
            },
            {
              vehicleType: VehicleTypeEnum.CAR,
              capId: '836154',
              manufacturerName: 'manufacturerName',
              rangeName: 'rangeName',
              derivativeName: 'derivativeName',
              averageRating: 4.5,
              isOnOffer: false,
              offerPosition: 5,
              leadTime: '',
              imageUrl: '',
              keyInformation: [],
              businessRate: 55,
              personalRate: 55,
            },
            {
              vehicleType: VehicleTypeEnum.CAR,
              capId: '836155',
              manufacturerName: 'manufacturerName',
              rangeName: 'rangeName',
              derivativeName: 'derivativeName',
              averageRating: 4.5,
              isOnOffer: false,
              offerPosition: 5,
              leadTime: '',
              imageUrl: '',
              keyInformation: [],
              businessRate: 55,
              personalRate: 55,
            },
            {
              vehicleType: VehicleTypeEnum.CAR,
              capId: '836156',
              manufacturerName: 'manufacturerName',
              rangeName: 'rangeName',
              derivativeName: 'derivativeName',
              averageRating: 4.5,
              isOnOffer: false,
              offerPosition: 5,
              leadTime: '',
              imageUrl: '',
              keyInformation: [],
              businessRate: 55,
              personalRate: 55,
            },
          ],
          derivatives: mockData.data.derivatives,
        },
        vehicleList: {
          edges: [],
        },
        refetch: jest.fn(),
      };
    },
  },
  {
    request: {
      query: GET_PRODUCT_CARDS_DATA,
      variables: {
        capIds: ['836151', '836152', '836153', '836154', '836155', '836156'],
        vehicleType: VehicleTypeEnum.CAR,
      },
    },
    result: () => {
      return {
        data: {
          productCard: mockData.data.productCard,
          derivatives: mockData.data.derivatives,
        },
        refetch: jest.fn(),
      };
    },
  },
];

describe('<SpecialOffersSearchContainer />', () => {
  beforeEach(async () => {
    await preloadAll();

    jest.clearAllMocks();
    filterMockCalled = false;
    window.sessionStorage.setItem = jest.fn();
  });

  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });

  it('should make a server request after render', async () => {
    // ACT
    act(() => {
      render(
        <MockedProvider mocks={mocksResponse} addTypename={false}>
          <SpecialOffersSearchContainer
            isCarSearch
            isServer={false}
            metaData={metaData}
          />
        </MockedProvider>,
      );
    });

    // ASSERT
    await waitFor(() => {
      expect(filterMockCalled).toBeTruthy();
    });
  });

  it('should be render correct list length', async () => {
    // ACT
    act(() => {
      render(
        <MockedProvider mocks={mocksResponse} addTypename={false}>
          <SpecialOffersSearchContainer isCarSearch metaData={metaData} />
        </MockedProvider>,
      );
    });

    // ASSERT
    await waitFor(() => {
      expect(filterMockCalled).toBeTruthy();
      expect(screen.getByText('Showing 0 Results')).toBeTruthy();
    });
  });

  it('should be render correctly', async () => {
    // ACT
    const getComponent = render(
      <MockedProvider mocks={mocksResponse} addTypename={false}>
        <SpecialOffersSearchContainer metaData={metaData} isCarSearch />
      </MockedProvider>,
    );

    await waitFor(() => {
      expect(screen.getByText('Automatic')).toBeInTheDocument();
    });
    const tree = getComponent.baseElement;
    expect(tree).toMatchSnapshot();
  });
});
