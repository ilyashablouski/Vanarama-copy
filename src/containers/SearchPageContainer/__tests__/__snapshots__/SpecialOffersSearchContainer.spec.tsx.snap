// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpecialOffersSearchContainer /> should be render correctly 1`] = `
<body>
  <div>
    <div
      class="row:title"
    >
      <nav
        class="breadcrumbs breadcrumbs--mobile"
        data-testid="breadcrumbs"
      >
        <ul
          class="breadcrumbs__list"
        >
          <li
            class="breadcrumbs__item"
          >
            <a
              class="breadcrumbs__link link -teal -small"
              data-testid="router-link"
              href="/"
            >
              <i
                class="icon -teal"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="328 112 184 256 328 400"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                  />
                </svg>
              </i>
              Back to 
              Home
            </a>
          </li>
        </ul>
      </nav>
      <nav
        class="breadcrumbs breadcrumbs--desktop"
        data-testid="breadcrumbs"
      >
        <ul
          class="breadcrumbs__list"
        >
          <li
            class="breadcrumbs__item"
          >
            <a
              class="breadcrumbs__link link -teal -small"
              data-testid="router-link"
              href="/"
            >
              Home
            </a>
            <i
              class="icon -medium -xsmall"
            >
              <svg
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
              >
                <polyline
                  points="184 112 328 256 184 400"
                  style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                />
              </svg>
            </i>
          </li>
          <li
            class="breadcrumbs__item"
          >
            <span
              class="text breadcrumbs__text -small -darker"
            >
              Car Leasing
            </span>
          </li>
        </ul>
      </nav>
      <h1
        class="heading -black -xlarge"
        data-uitestid="undefined_page-title_heading"
      >
        HubCarPage
         
        <br />
      </h1>
    </div>
    <div
      class="row:bg-light -xthin"
    >
      <div
        class="row:search-filters"
      >
        <div
          class="search-filters"
        >
          <div
            aria-level="3"
            class="search-filters--title"
            role="heading"
          >
            <i
              class="icon search-filters--title-icon"
            >
              <svg
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M64,144H290.75a48,48,0,0,0,90.5,0H448a16,16,0,0,0,0-32H381.25a48,48,0,0,0-90.5,0H64a16,16,0,0,0,0,32Z"
                />
                <path
                  d="M448,368H381.25a48,48,0,0,0-90.5,0H64a16,16,0,0,0,0,32H290.75a48,48,0,0,0,90.5,0H448a16,16,0,0,0,0-32Z"
                />
                <path
                  d="M448,240H221.25a48,48,0,0,0-90.5,0H64a16,16,0,0,0,0,32h66.75a48,48,0,0,0,90.5,0H448a16,16,0,0,0,0-32Z"
                />
              </svg>
            </i>
            <span>
              Filters
            </span>
            <i
              class="icon search-filters--title-icon -white"
            >
              <svg
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
              >
                <polyline
                  points="112 184 256 328 400 184"
                  style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                />
              </svg>
            </i>
          </div>
          <div
            class="toggle--wrapper search-filters--toggle"
          >
            <input
              autocomplete="off"
              checked=""
              class="toggle toggle--input"
              id="contractType"
              type="checkbox"
            />
            <label
              class="toggle--input-label -orange"
              for="contractType"
            >
              <span
                class="toggle--switch"
              >
                <span
                  class="toggle--on"
                >
                  Personal
                </span>
                <span
                  class="toggle--off"
                >
                  Business
                </span>
              </span>
            </label>
          </div>
          <div
            class="dropdown search-filters--dropdown"
          >
            <button
              class="dropdown--toggle"
              type="button"
            >
              <span>
                Make & Model
              </span>
              <i
                class="icon -dark"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="112 184 256 328 400 184"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
            </button>
            <div
              class="dropdown--content"
            >
              <div
                class="formgroup"
              >
                <label
                  class="formgroup--legend"
                >
                  Make
                </label>
                <div
                  class="formgroup--content"
                >
                  <div
                    class="select"
                  >
                    <select
                      class="select--native"
                      name="manufacturer"
                    >
                      <option
                        disabled=""
                        value=""
                      >
                        Select make
                      </option>
                      <option
                        value="Citroën"
                      >
                        Citroën
                      </option>
                      <option
                        value="Dacia"
                      >
                        Dacia
                      </option>
                      <option
                        value="BMW"
                      >
                        BMW
                      </option>
                    </select>
                    <span
                      class="icon select--chevron"
                    >
                      <i
                        class="icon -stroke -dark"
                      >
                        <svg
                          height="1em"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <polyline
                            points="112 184 256 328 400 184"
                            style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="formgroup"
              >
                <label
                  class="formgroup--legend"
                >
                  Model
                </label>
                <div
                  class="formgroup--content"
                >
                  <div
                    class="select"
                  >
                    <select
                      class="select--native"
                      name="model"
                    >
                      <option
                        disabled=""
                        value=""
                      >
                        Select model
                      </option>
                    </select>
                    <span
                      class="icon select--chevron"
                    >
                      <i
                        class="icon -stroke -dark"
                      >
                        <svg
                          height="1em"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <polyline
                            points="112 184 256 328 400 184"
                            style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </div>
              <button
                class="button -fullwidth -teal -small -solid"
                data-testid="Make & Modelbtn"
                data-uitestid="undefined_button_Make & Model"
              >
                <div
                  class="button--inner"
                >
                  View 0 Results
                </div>
              </button>
            </div>
          </div>
          <div
            class="dropdown search-filters--dropdown"
          >
            <button
              class="dropdown--toggle"
              type="button"
            >
              <span>
                Budget
              </span>
              <i
                class="icon -dark"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="112 184 256 328 400 184"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
            </button>
            <div
              class="dropdown--content"
            >
              <div
                class="formgroup"
              >
                <label
                  class="formgroup--legend"
                >
                  Price From
                </label>
                <div
                  class="formgroup--content"
                >
                  <div
                    class="select"
                  >
                    <select
                      class="select--native"
                      name="from"
                    >
                      <option
                        disabled=""
                        value=""
                      >
                        Select from
                      </option>
                      <option
                        value="0"
                      >
                        £0
                      </option>
                      <option
                        value="150"
                      >
                        £150
                      </option>
                      <option
                        value="250"
                      >
                        £250
                      </option>
                      <option
                        value="350"
                      >
                        £350
                      </option>
                      <option
                        value="450"
                      >
                        £450
                      </option>
                      <option
                        value="550"
                      >
                        £550
                      </option>
                    </select>
                    <span
                      class="icon select--chevron"
                    >
                      <i
                        class="icon -stroke -dark"
                      >
                        <svg
                          height="1em"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <polyline
                            points="112 184 256 328 400 184"
                            style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="formgroup"
              >
                <label
                  class="formgroup--legend"
                >
                  Price To
                </label>
                <div
                  class="formgroup--content"
                >
                  <div
                    class="select"
                  >
                    <select
                      class="select--native"
                      name="to"
                    >
                      <option
                        disabled=""
                        value=""
                      >
                        Select to
                      </option>
                      <option
                        value="150"
                      >
                        £150
                      </option>
                      <option
                        value="250"
                      >
                        £250
                      </option>
                      <option
                        value="350"
                      >
                        £350
                      </option>
                      <option
                        value="450"
                      >
                        £450
                      </option>
                      <option
                        value="550"
                      >
                        £550
                      </option>
                      <option
                        value="550+"
                      >
                        £550+
                      </option>
                    </select>
                    <span
                      class="icon select--chevron"
                    >
                      <i
                        class="icon -stroke -dark"
                      >
                        <svg
                          height="1em"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <polyline
                            points="112 184 256 328 400 184"
                            style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                          />
                        </svg>
                      </i>
                    </span>
                  </div>
                </div>
              </div>
              <button
                class="button -fullwidth -teal -small -solid"
                data-testid="Budgetbtn"
                data-uitestid="undefined_button_Budget"
              >
                <div
                  class="button--inner"
                >
                  View 0 Results
                </div>
              </button>
            </div>
          </div>
          <div
            class="dropdown search-filters--dropdown"
          >
            <button
              class="dropdown--toggle"
              type="button"
            >
              <span>
                Body Type
              </span>
              <i
                class="icon -dark"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="112 184 256 328 400 184"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
            </button>
            <div
              class="dropdown--content"
            >
              <div
                class="formgroup"
                data-testid="Body Type"
                data-uitestid="undefined_formgroup_Body Type"
              >
                <label
                  class="formgroup--legend"
                >
                  Body Type
                </label>
                <div
                  class="formgroup--content"
                >
                  <div
                    class="choice-boxes -cols-1 -medium"
                  >
                    <button
                      class="choice-box"
                      type="button"
                    >
                      <span
                        class="choice-label"
                      >
                        Dropside Tipper
                      </span>
                    </button>
                    <button
                      class="choice-box"
                      type="button"
                    >
                      <span
                        class="choice-label"
                      >
                        Large Van
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <button
                class="button -fullwidth -teal -small -solid"
                data-testid="Body Typebtn"
                data-uitestid="undefined_button_Body Type"
              >
                <div
                  class="button--inner"
                >
                  View 0 Results
                </div>
              </button>
            </div>
          </div>
          <div
            class="dropdown search-filters--dropdown"
          >
            <button
              class="dropdown--toggle"
              type="button"
            >
              <span>
                Transmission
              </span>
              <i
                class="icon -dark"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="112 184 256 328 400 184"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
            </button>
            <div
              class="dropdown--content"
            >
              <div
                class="formgroup"
                data-testid="Transmission"
                data-uitestid="undefined_formgroup_Transmission"
              >
                <label
                  class="formgroup--legend"
                >
                  Transmission
                </label>
                <div
                  class="formgroup--content"
                >
                  <div
                    class="choice-boxes -cols-1 -medium"
                  >
                    <button
                      class="choice-box"
                      type="button"
                    >
                      <span
                        class="choice-label"
                      >
                        Automatic
                      </span>
                    </button>
                    <button
                      class="choice-box"
                      type="button"
                    >
                      <span
                        class="choice-label"
                      >
                        Manual
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <button
                class="button -fullwidth -teal -small -solid"
                data-testid="Transmissionbtn"
                data-uitestid="undefined_button_Transmission"
              >
                <div
                  class="button--inner"
                >
                  View 0 Results
                </div>
              </button>
            </div>
          </div>
          <div
            class="dropdown search-filters--dropdown"
          >
            <button
              class="dropdown--toggle"
              type="button"
            >
              <span>
                Fuel Type
              </span>
              <i
                class="icon -dark"
              >
                <svg
                  height="1em"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <polyline
                    points="112 184 256 328 400 184"
                    style="fill: none; stroke: var(--class-color); stroke-linecap: square; stroke-miterlimit: 10; stroke-width: 48;"
                  />
                </svg>
              </i>
            </button>
            <div
              class="dropdown--content"
            >
              <div
                class="formgroup"
                data-testid="Fuel Type"
                data-uitestid="undefined_formgroup_Fuel Type"
              >
                <label
                  class="formgroup--legend"
                >
                  Fuel Type
                </label>
                <div
                  class="formgroup--content"
                >
                  <div
                    class="choice-boxes -cols-1 -medium"
                  >
                    <button
                      class="choice-box"
                      type="button"
                    >
                      <span
                        class="choice-label"
                      >
                        diesel
                      </span>
                    </button>
                    <button
                      class="choice-box"
                      type="button"
                    >
                      <span
                        class="choice-label"
                      >
                        iii
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <button
                class="button -fullwidth -teal -small -solid"
                data-testid="Fuel Typebtn"
                data-uitestid="undefined_button_Fuel Type"
              >
                <div
                  class="button--inner"
                >
                  View 0 Results
                </div>
              </button>
            </div>
          </div>
          <div
            class="search-filters--tags"
          />
        </div>
      </div>
    </div>
    <div
      class="row:bg-lighter -thin"
    >
      <div
        class="row:results"
      >
        <span
          class="text -regular -darker"
          data-uitestid="undefined_text_results-count"
        >
          Showing 0 Results
        </span>
        <div
          class="select"
        >
          <select
            class="select--native"
          >
            <option
              disabled=""
              value=""
            >
              Please Select
            </option>
            <option
              data-uitestid="undefined_rate_ASC"
              value="rate_ASC"
            >
              Price Low To High
            </option>
            <option
              data-uitestid="undefined_rate_DESC"
              value="rate_DESC"
            >
              Price High To Low
            </option>
            <option
              data-uitestid="undefined_availability_ASC"
              value="availability_ASC"
            >
              Fastest Delivery Time
            </option>
          </select>
          <span
            class="icon select--chevron"
          >
            <i
              class="icon -stroke -dark"
            >
              <svg
                height="1em"
                viewBox="0 0 512 512"
                width="1em"
              >
                <polyline
                  points="112 184 256 328 400 184"
                  style="fill: none; stroke: var(--class-color); stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                />
              </svg>
            </i>
          </span>
        </div>
        <div
          class="row:cards-3col"
        />
        <div
          class="pagination"
        />
      </div>
    </div>
    <div
      class="row:text"
    >
      <span
        class="text -regular -dark"
      >
        Photos and videos are for illustration purposes only.
         
        <a
          class="link -teal"
          data-testid="router-link"
          data-uitestid="undefined_link_terms-and-conditions"
          href="/legal/terms-and-conditions"
        >
          Terms and conditions apply
        </a>
        .
      </span>
    </div>
    <script
      type="application/ld+json"
    >
      {"@graph":[{"@type":"FAQPage","mainEntity":[{"name":"Why Lease A Car?","@type":"Question","acceptedAnswer":{"text":"Leasing your brand new car with Vanarama means you can drive the car of your dreams at a price you can afford. And you can upgrade to a new one every few years","@type":"Answer"}},{"name":"How Does Car Leasing Work?","@type":"Question","acceptedAnswer":{"text":"Personal car leasing works very much like renting, but for a longer time period. You choose your new car, the contract length, how much you'd like your initial rental to be and how many miles you'll drive over the period. At the end of your lease agreement, you return the car & upgrade","@type":"Answer"}}]},{"url":"https://www.vanarama.com/car-leasing.html","name":"Vanarama","@type":"Service","@context":"http://schema.org","description":"Leasing your brand new car with Vanarama means you can drive the car of your dreams at a price you can afford.","serviceType":"Car Leasing"}]}
    </script>
  </div>
</body>
`;
